Test/Unit/Helper/Webhook/CaptureWebhookHandlerTest.php

```diff
@@ -0,0 +1,199 @@
+
+        // Mock the adyenOrderPaymentFactory
+        $adyenOrderPaymentFactoryMock = $this->createGeneratedMock(PaymentFactory::class, ['create']);
+
+        $adyenOrderPaymentMock = $this->getMockBuilder(Payment::class)
+            ->setMethods(['load']) // Define the method you want to mock
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $adyenOrderPaymentMock->expects($this->once())
+            ->method('load')
+            ->with(123, OrderPaymentInterface::ENTITY_ID)
+            ->willReturnSelf(); // Return the mock itself
+
+        // Set up expectations for the create and load methods
+        $adyenOrderPaymentFactoryMock->expects($this->once())
+            ->method('create')
+            ->willReturn($adyenOrderPaymentMock);
+
+        $adyenOrderPaymentHelperMock = $this->createMock(AdyenOrderPayment::class);
+
+        $adyenOrderPaymentHelperMock->expects($this->once())
+            ->method('refreshPaymentCaptureStatus')
+            ->with($adyenOrderPaymentMock, $this->notification->getAmountCurrency());
+
+        // Create a mock for the magentoInvoiceFactory
+        $magentoInvoiceFactoryMock = $this->createGeneratedMock(MagentoInvoiceFactory::class, ['create']);
+
+        // Create a mock for MagentoInvoice
+        $magentoInvoiceMock = $this->getMockBuilder(MagentoInvoice::class)
+            ->setMethods(['load']) // Define the method you want to mock
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        // Configure the load method of the magentoInvoiceMock to return the same mock
+        $magentoInvoiceMock->expects($this->once())
+            ->method('load')
+            ->with(456, MagentoInvoice::ENTITY_ID)
+            ->willReturnSelf(); // Return the mock itself
+
+        // Configure the magentoInvoiceFactoryMock to return the magentoInvoiceMock
+        $magentoInvoiceFactoryMock->expects($this->once())
+            ->method('create')
+            ->willReturn($magentoInvoiceMock);
+
+        $this->captureWebhookHandler = $this->createCaptureWebhookHandler(
+            $invoiceHelperMock,
+            $adyenOrderPaymentFactoryMock,
+            $adyenOrderPaymentHelperMock,
+            null,
+            $magentoInvoiceFactoryMock,
+            $orderHelperMock,
+            $paymentMethodsHelperMock)
+        ;
+
+        // Test handleWebhook method
+        $result = $this->captureWebhookHandler->handleWebhook($this->order, $this->notification, 'paid');
+
+        // Assert that the order is finalized
+        $this->assertSame($this->order, $result);
+    }
+
+    public function testHandleWebhookTransitionNotPaid()
+    {
+        // Test handleWebhook method with transition state different from "PAID"
+        $result = $this->captureWebhookHandler->handleWebhook($this->order, $this->notification, 'NOT_PAID');
+
+        // Assert that the order is not modified
+        $this->assertSame($this->order, $result);
```

Comment:
Better to assert with https://www.geeksforgeeks.org/phpunit-assertequalscanonicalizing-function/
```suggestion
        $this-> assertEqualsCanonicalizing($this->order, $result);
```